src/mumble/pluginAPI/Plugin.h

```diff
@@ -182,7 +182,7 @@ extern "C" {
 			MumbleChannelID_t newChannelID);
 
 	/// Called whenever a user leaves a channel.
-	/// This function will also be called when the user disconnects from the server.
+	/// This includes a client disconnecting from the server as this will also lead to the user not being in that channel anymore.
```

Comment:
So to differentiate between "user disconnected" and "user switched channel" we have to store the channel leave, then wait a unspecified amount of time and check if onChannelEntered is called to see if the client is still there.
Or have to retrieve all clients of server to check if client is still on server.

Reply:
I originally intended to add callbacks for generic user connect/disconnect handlers but I couldn't find a nice position in Mumble's code to hook into. That's why I haven't added them yet.

Maybe I'll find a suitable way to do so when implementing the framework but for now there's no callback for that. Not pretty but for a first version I guess this might be forgivable.